### PR TITLE
feat: metrics agent

### PR DESCRIPTION
**Pull request by patrykcieszkowski**

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -7,6 +7,9 @@
     "interval": 300,
     "maxDeviation": 300
   },
+  "stats": {
+    "interval": 10
```
Comment (jimaek) [inline config/default.json:11]: How bad would it be if we lowered it even further? To get more up to date data

Comment (patrykcieszkowski) [reply config/default.json:11]: this won't be an issue for the probe, but I don't know how well the server would handle it. I guess we gotta check and see.

File: src/lib/stats/client.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		jobs: {
+			count: jobCount,
+		},
+	});
+};
+
+export const run = (socket: Socket, worker: Worker) => {
+	setInterval(async () => {
+		await report(socket, worker.jobs.size);
+	}, statsConfig.interval * 1000);
```
Comment (MartinKolarik) [inline src/lib/stats/client.ts:25]:
```suggestion
	setInterval(() => {
		report(socket, worker.jobs.size).catch((error) => { /* TODO: log? */ });
	}, statsConfig.interval * 1000);
```

Similar to the recent probe handler issue - the interval function isn't supposed to return a promise, and doing so makes it seem as if the potential errors were handled, but they aren't (I guess here nothing can really throw but still, this is a bad pattern).

Comment (patrykcieszkowski) [reply src/lib/stats/client.ts:25]: That's a stretch. There's no way this method would ever throw an error, and attaching an error handler to it only adds a layer of unecessary complexity.

**Pull request merged**